diff --git a/gcc/jit/Make-lang.in b/gcc/jit/Make-lang.in
index eb1c996..59497bd 100644
--- a/gcc/jit/Make-lang.in
+++ b/gcc/jit/Make-lang.in
@@ -73 +73 @@ LIBGCCJIT_SONAME = \
-LIBGCCJIT_FILENAME = $(LIBGCCJIT_BASENAME).$(LIBGCCJIT_VERSION_NUM).dylib
+LIBGCCJIT_FILENAME = $(LIBGCCJIT_BASENAME).a
@@ -93,2 +92,0 @@ jit: $(LIBGCCJIT_FILENAME) \
-	$(LIBGCCJIT_SYMLINK) \
-	$(LIBGCCJIT_LINKER_NAME_SYMLINK) \
@@ -189,0 +188,7 @@ $(LIBGCCJIT_FILENAME): $(jit_OBJS) \
+ifneq (,$(findstring darwin,$(host)))
+	rm -f $@
+	/usr/bin/libtool -static -o $@ \
+	     $(jit_OBJS) libbackend.a libcommon-target.a libcommon.a \
+	     $(CPPLIB) $(LIBDECNUMBER) $(LIBBACKTRACE) $(LIBIBERTY) \
+	     $(EXTRA_GCC_OBJS_EXCLUSIVE)
+else
@@ -194,0 +200 @@ $(LIBGCCJIT_FILENAME): $(jit_OBJS) \
+endif
@@ -398 +404 @@ jit.install-common: installdirs jit.install-headers
-	$(INSTALL_PROGRAM) $(LIBGCCJIT_FILENAME) \
+	$(INSTALL_DATA) $(LIBGCCJIT_FILENAME) \
@@ -400,3 +405,0 @@ jit.install-common: installdirs jit.install-headers
-	ln -sf \
-	  $(LIBGCCJIT_SONAME_SYMLINK)\
-	  $(DESTDIR)$(libdir)/$(LIBGCCJIT_LINKER_NAME_SYMLINK)
